### PR TITLE
fix(discover) Display tags that look like function aggregates

### PR DESCRIPTION
**Pull request by markstory**
When using a tag like `count_diff` the API endpoint would incorrectly hint that the output type was an integer when the value was a string. This was caused by a greedy pattern in the metadata formatter. We shouldonly retype values that are not strings in snuba/clickhouse.

### CODE REVIEW

File: src/sentry/api/bases/organization_events.py [hunk 1]
```diff
@@ -148,6 +148,9 @@ def handle_results_with_meta(self, request, organization, project_ids, results):
             if not data:
                 return {"data": [], "meta": {}}
 
+            import logging
```
Comment (evanh) [inline src/sentry/api/bases/organization_events.py:151]: Pretty sure this should be removed.

File: tests/sentry/api/test_event_search.py [hunk 2]
```diff
@@ -41,18 +41,21 @@ def test_get_json_meta_type():
...
+    assert get_json_meta_type("p75", "Float32") == "duration"
+    assert get_json_meta_type("p95", "Float32") == "duration"
+    assert get_json_meta_type("p99", "Float32") == "duration"
+    assert get_json_meta_type("p100", "Float32") == "duration"
+    assert get_json_meta_type("apdex_transaction_duration_300", "Float32") == "number"
+    assert get_json_meta_type("failure_rate", "Float32") == "percentage"
+    assert get_json_meta_type("impact_300", "Float32") == "number"
+    assert get_json_meta_type("user_misery_300", "Float32") == "number"
+    assert get_json_meta_type("percentile_transaction_duration_0_95", "Float32") == "duration"
+    assert get_json_meta_type("count_thing", "UInt64") == "integer"
```
Comment (evanh) [inline tests/sentry/api/test_event_search.py:56]: What are these for?

File: tests/snuba/api/endpoints/test_organization_events_v2.py [hunk 3]
```diff
@@ -1839,6 +1839,33 @@ def test_not_has_transaction_status(self):
...
+                    "query": "",
+                    "project": [project.id],
+                    "statsPeriod": "24h",
+                },
+            )
+            assert response.status_code == 200, response.content
+            meta = response.data["meta"]
+            assert "string" == meta["count_diff"], "tags should not be counted as integers"
+            assert "string" == meta["message"]
+            assert "integer" == meta["count"]
```
Comment (evanh) [inline tests/snuba/api/endpoints/test_organization_events_v2.py:1867]: Probably worth it to check the value of the tag as well.

File: src/sentry/api/bases/organization_events.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/bases/organization_events.py:151]: Whoops.

File: tests/sentry/api/test_event_search.py (hunk 2 above)
Comment (markstory) [reply tests/sentry/api/test_event_search.py:56]: This is to make sure that the count function still returns an integer. The previous tests has the types wrong as the second parameter should be a snuba/clickhouse type and not the output types.

**Review by dashed — APPROVED** [review]

**Pull request merged**